lib/Router.js

```diff
@@ -114,7 +144,11 @@ function documentRoute(route) {
 Router.prototype.status = function(code, body) {
     body = body || {};
     body.$httpStatus = parseInt(code, 10);
-    return Promise.resolve(body);
+    if (code >= 200 && code < 300) {
```

Comment:
Good question. My original thought was that only 2xx "success" status codes would qualify for a success handler. But it might be reasonable for 304 to be considered a success a well.

It should be up to clients (and the internal API call functionality) to handle 3xx responses accordingly -- by transparently following redirects or restoring from cache (in the case of 304).

I can update this as soon as 3xx codes are needed. This will be useful for supporting collection caches, but not sure if we'll need it before that is implemented.